src/express-server.ts

```diff
@@ -0,0 +1,229 @@
+import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
+import { StreamableHTTPServerTransport } from "@modelcontextprotocol/sdk/server/streamableHttp.js";
+import {
+  InitializeRequest,
+  InitializeRequestSchema,
+} from "@modelcontextprotocol/sdk/types.js";
+import { randomUUID } from "crypto";
+import express, { NextFunction, Request, Response } from "express";
+import { FireHydrantCore } from "./core.js";
+import { createConsoleLogger } from "./mcp-server/console-logger.js";
+import { createMCPServer } from "./mcp-server/server.js";
+
+type Props = Record<string, string>;
+
+// Constants
+const HEADERS = {
+  AUTHORIZATION: "authorization",
+  SESSION_ID: "mcp-session-id",
+} as const;
+
+const DEFAULT_PORT = 3000;
+const LOG_LEVEL = "debug";
+
+const ERROR_MESSAGES = {
+  AUTH_REQUIRED: "Authorization header is required",
+  AUTH_EMPTY: "Authorization header cannot be empty",
+  SESSION_REQUIRED: "mcp-session-id header is required",
+  TRANSPORT_NOT_FOUND: "Transport not found",
+  INVALID_SESSION: "Invalid session ID or method",
+  INTERNAL_ERROR: "Internal server error",
+} as const;
+
+class HTTPError extends Error {
+  status: number;
+
+  constructor(status: number, message: string) {
+    super(message);
+    this.status = status;
+    this.name = "HTTPError";
+  }
+}
+
+// Extended Express Request interface
+declare global {
+  namespace Express {
+    interface Request {
+      ctx: Props;
+      mcpServer: McpServer;
+    }
+  }
+}
+
+// Transport storage
+type TransportMap = Map<string, StreamableHTTPServerTransport>;
+
+const httpTransportMap: TransportMap = new Map();
```

Comment:
is it possible to deploy multiple and get routed to the wrong one load balancing?

Reply:
since session management is optional, i've gone ahead and removed this.